gr-dtv/include/gnuradio/dtv/atsc_plinfo.h

```diff
@@ -0,0 +1,151 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2001,2006,2014 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifndef DTV_INCLUDED_ATSC_PLINFO_H
+#define DTV_INCLUDED_ATSC_PLINFO_H
+
+#include <gnuradio/dtv/atsc_consts.h>
+#include <cassert>
+#include <cstring>
+
+namespace gr {
+namespace dtv {
+
+/*!
+ * \brief pipeline info that flows with data
+ *
+ * Not all modules need all the info
+ */
+class plinfo
+{
+public:
+    plinfo() : _flags(0), _segno(0) {}
+    // plinfo(uint16_t flags_, int16_t segno_) : _flags(flags_), _segno(segno_) {}
+
+    // accessors
+
+    bool field_sync1_p() const { return (_flags & fl_field_sync1) != 0; }
```

Comment:
Since this class is now part of the public API (and even ABI) it could make sense to clean it up a bit.

E.g.:
* Add `noexcept` where appropriate
* change state machine (?) to `enum class`
* use member initialization
* remove `operator==/!=` (they're default, no?
* Make member variables private?
* Put implementation in a .cc in `lib/`

What do you think?

Reply:
Agreed - I have more work to clean up this class

Reply:
Ok, I've done the things here except for the `noexcept` - would like to see this applied consistently through the codebase and it is currently mainly used for `noexcept(false)`

Reply:
There's something with `noexcept` and new C++ standards. Something changed -- @ThomasHabets would know best.

Reply:
`noexcept(false)` is the default. I'm not sure if you're thinking of the now deprecated `throw()` spec. Except when writing templates that can instantiatie to `noexcept` or not depending on template args, there's no need for parentheses.

`throw()` was always a mess, unfortunately. `noexcept` is fine though. `throw()` ended up being only used for empty throw specs.

In my testing `noexcept` reduces code size. I've not benchmarked but it can enable other optimizations beyond reduced instruction cache use.

But the main reason I suggested it is that it's a nice annotation to the reader that this function cannot even in principle throw. I'm pretty sure it's not part of the ABI, but for public APIs better safe than sorry so should only be added when no possible future implementation could throw, in my opinion.

Of course in this case it'll in practice be inlined anyway, and the compiler will not generate exception handling code on entry or exit.

Reply:
I definitely trust @ThomasHabets on this one.  Pushed a commit that puts `noexcept` _everywhere_ in the plinfo class

Reply:
On second thought, I don't think we're ready to move in the direction of putting `noexcept` in lots of places, and probably need a consistent strategy for exceptions and exception handling.  Backing out the last commit to keep this file consistent with the rest of the codebase

Reply:
Thinking first and rolling out a consistent use based on clear criteria sounds best, I agree.